source/lac/trilinos_sparse_matrix.cc

```diff
@@ -695,14 +707,11 @@ namespace TrilinosWrappers
       // finalize nonlocal graph and create nonlocal matrix
       if (nonlocal_graph.get() != 0)
         {
-          if (nonlocal_graph->IndicesAreGlobal() == false &&
-              nonlocal_graph->RowMap().NumMyElements() > 0)
-            {
-              // insert dummy element
-              TrilinosWrappers::types::int_type row =
-                nonlocal_graph->RowMap().MyGID(TrilinosWrappers::types::int_type(0));
-              nonlocal_graph->InsertGlobalIndices(row, 1, &row);
-            }
+          // must make sure the IndicesAreGlobal flag is set on all processors
+          // because some processors might not call InsertGlobalIndices (and
+          // we do not want to insert dummy indices on all processors for
+          // large-scale simulations due to the bad impact on performance)
+          nonlocal_graph->SetIndicesAreGlobal();
```

Comment:
Unfortunately not (at least I do not see how I could do it). The same code is present in Epetra_FECrsGraph, but there access is possible because it is derived from Epetra_CrsGraph, so I had to be a bit creative here. Not nice but works.